contracts/FluxPriceFeedFactory.sol

```diff
@@ -0,0 +1,134 @@
+
+    // mapping of id to FluxPriceFeed
+    mapping(bytes32 => FluxPriceFeed) public fluxPriceFeeds;
+
+    event FluxPriceFeedCreated(bytes32 indexed id, address indexed oracle);
+    event Log(string message);
+
+    constructor(address _validator) {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(VALIDATOR_ROLE, _validator);
+    }
+
+    /**
+     * @notice internal function to create a new FluxPriceFeed
+     * @dev only a validator should be able to call this function
+     */
+    function _deployOracle(
+        bytes32 _id,
+        string calldata _pricePair,
+        uint8 _decimals
+    ) internal {
+        // deploy the new contract and store it in the mapping
+        FluxPriceFeed newPriceFeed = new FluxPriceFeed(address(this), _decimals, _pricePair);
+        fluxPriceFeeds[_id] = newPriceFeed;
+
+        // also grant this contract's admin VALIDATOR_ROLE on the new FluxPriceFeed
+        newPriceFeed.grantRole(VALIDATOR_ROLE, msg.sender);
+        console.logBytes32(_id);
+        console.log(address(newPriceFeed));
+        emit FluxPriceFeedCreated(_id, address(newPriceFeed));
+    }
+
+    /**
+     * @notice transmit submits an answer to a price feed or creates a new one if it does not exist
+     * @param _pricePairs array of price pairs strings (e.g. ETH/USD)
+     * @param _decimals array of decimals for associated price pairs (e.g. 3)
+     * @param _answers array of prices for associated price pairs
+     */
+    function transmit(
+        string[] calldata _pricePairs,
+        uint8[] calldata _decimals,
+        int192[] calldata _answers
+    ) external onlyRole(VALIDATOR_ROLE) {
+        require(
+            (_pricePairs.length == _decimals.length) && (_pricePairs.length == _answers.length),
+            "Transmitted arrays must be equal"
+        );
+        // Iterate through each transmitted price pair
+        for (uint256 i = 0; i < _pricePairs.length; i++) {
+            // Find the price pair id
+            string memory str = string(abi.encodePacked("Price-", _pricePairs[i], "-", Strings.toString(_decimals[i])));
+            bytes32 id = keccak256(bytes(str));
+
+            // if oracle exists then transmit values
+            if (address(fluxPriceFeeds[id]) != address(0x0)) {
+                // try to transmit or create new price pair if it doesn't exist
+                try fluxPriceFeeds[id].transmit(_answers[i]) {
+                    // transmission is successful, nothing to do
+                } catch Error(string memory reason) {
+                    // catch failing revert() and require()
+                    console.log("ERRORR");
+                    emit Log(reason);
+                }
+                // else create an oracle then transmit values
+            } else {
+                _deployOracle(id, _pricePairs[i], _decimals[i]);
+                fluxPriceFeeds[id].transmit(_answers[i]);
```

Comment:
I wonder if we should change the logic of this `if-else` statements to unify the `transmit` call.

```solidity
if (address(fluxPriceFeeds[id]) == address(0x0) {
    _deployOracle(id, _pricePairs[i], _decimals[i]);
}

try fluxPriceFeeds[id].transmit(_answers[i]) {
    // transmission is successful, nothing to do
} catch Error(string memory reason) {
    // catch failing revert() and require()
    emit Log(reason);
}
```

Reply:
Just did that!